tests/main/apparmor-prompting-flag-restart/task.yaml

```diff
@@ -0,0 +1,121 @@
+summary: Check that changing the experimental.apparmor-prompting flag causes snapd to restart
+
+details: |
+    This test checks that snapd is restarted when the experimental.apparmor-prompting
+    flag is changed, and that snapd is not restarted when the flag is set to
+    the same value repeatedly. Restarting snapd when the flag changes is
+    necessary so that apparmor kernel and parser features are re-probed, in
+    case support for prompting has changed since snapd last started.
+
+systems:
+  - ubuntu-16.04-*
+  - ubuntu-18.04-*
+  - ubuntu-2*
+
+prepare: |
+    if ! command -v curl; then
+        snap install --devmode --edge test-snapd-curl
+        snap alias test-snapd-curl.curl curl
+    fi
+
+debug: |
+    # Report system-info
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/system-info | jq
+
+execute: |
+    . /etc/os-release
+
+    # Necessary since we restart snapd many times
+    restart_snapd() {
+        systemctl stop snapd.service snapd.socket
+        systemctl reset-failed snapd.service snapd.socket
```

Comment:
I wonder why this get increased, as the snapd service exits itself using this config:

SuccessExitStatus=42
RestartPreventExitStatus=42

in the service to avoid its own restart to count as failures?  @valentindavid maybe you know more about this/what is happening here

Reply:
It should not. It is in a success status.

Reply:
OK, now I understand a bit more thanks to some explanation from @olivercalder

No matter the status, if a service starts more than `StartLimitBurst` in `StartLimitIntervalSec` (duration), then the service is failed.

The proper way to handle this properly in snapd would be:
 * Take the monotonic time at start of snapd
 * When we want to restart, wait until the clock has passed start + StartLimitIntervalSec/StartLimitBurst.

The default values for systemd are 10s for 5 start. So wait 2s before restart.

Reply:
We could test that `systemctl show -P Result snapd.service` gives `start-limit-hit` and restart. Then that would be safe to do in tests. And maybe add a script in tests/lib/`

Reply:
I'm a bit wary to attempt to introduce changes to the restart behavior of snapd (for `RestartDaemon` and `RestartSocket`, not `RestartSystem` or others) which are based on a default value for systemd. If we can look up these values at snapd startup that would be preferable, though I'm still wary, and this would likely be outside the scope of this PR.

Also, it seems like modifying `StartLimitIntervalSec` in tests is not advised. See:
```sh
$ git log --grep StartLimitBurst
$ git log --grep StartLimitIntervalSec
```

In particular:
- 116852479d508c438c8f29cfbcc97666d8c826a6
- 72cf413398635a61afd20b7f0e9f96e6c50e830d

Reply:
Of course, the change in reboot behavior (with the sleep) should be done in a different PR. It is a substantial change.

However in this PR you could verify that the state is really `start-limit-hit` before resetting the state.

And after this PR is merged we can also have a separate PR to refactor tests to use that check everywhere we have been doing `reset-failed`. Until we fixed the reboot behavior to wait.